projects/step-frontend/src/lib/modules/timeseries/metric-chart/metric-chart.component.ts

```diff
@@ -177,6 +219,45 @@ export class MetricChartComponent implements OnInit, OnChanges {
     };
   }
 
+  /**
+   * This function will fetch the entities found in the labels, and replace their ids with the appropriate name.
+   * @param series
+   * @param groupDimensionIndex
+   * @param fetchByIds
+   * @param mapEntityToLabel
+   * @private
+   */
+  private fetchAndSetLabels<T>(
+    series: TSChartSeries[],
+    groupDimensionIndex: number,
+    fetchByIds: (ids: string[]) => Observable<T[]>,
+    mapEntityToLabel: (entity: T) => string | undefined
+  ) {
+    const ids: string[] = series.map((s) => s.labelItems[groupDimensionIndex] as string).filter((id) => id);
+
+    if (ids.length > 0) {
```

Comment:
I would prefer to invert the condition and do a return, instead of adding additional nesting block
```
if (!ids.length) {
  return;
}
```